src/Common/Wpf/Impl/Collections/RangeObservableCollection.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+
+namespace Microsoft.Common.Wpf.Collections {
+    /// <summary>
+    /// ObservableCollection that supports bulk changes
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class RangeObservableCollection<T> : ObservableCollection<T> {
+
+        public RangeObservableCollection() { }
+
+        public RangeObservableCollection(IEnumerable<T> collection) : base(collection) { }
+
+        public void AddRange(IEnumerable<T> collection) {
+            if (collection == null) {
+                throw new ArgumentNullException(nameof(collection));
+            }
+
+            foreach (var i in collection) {
+                Items.Add(i);
```

Comment:
nvm, it seems that all calls are via

```
async Task SwitchToAvailablePackagesAsync() {
            _coreShell.AssertIsOnMainThread();
```

which confuses me since it a s async method and yet has to be called on UI thread...